tests/rest/test_connection.py

```diff
@@ -2456,6 +2456,88 @@ def test_properties(self, con120):
             }
         }
 
+    def test_load_stac_from_job_canonical(self, con120, requests_mock):
+        requests_mock.get(
+            API_URL + "jobs/j0bi6/results",
+            json={
+                "links": [
+                    {
+                        "href": "https://wrong.link",
+                        "rel": "self",
+                    },
+                    {
+                        "href": "https://stac.link",
+                        "rel": "canonical",
+                    },
+                ]
+            },
+        )
+        job = con120.job("j0bi6")
+        cube = con120.load_stac_from_job(job)
+
+        fg = cube.flat_graph()
+        assert fg == {
+            "loadstac1": {
+                "process_id": "load_stac",
+                "arguments": {"url": "https://stac.link"},
+                "result": True,
+            }
+        }
+
+    def test_load_stac_from_job_unsigned(self, con120, requests_mock):
+        requests_mock.get(
+            API_URL + "jobs/j0bi6/results",
+            json={
+                "links": [
+                    {
+                        "href": "https://wrong.link",
+                        "rel": "self",
+                    },
+                ]
+            },
+        )
+        job = con120.job("j0bi6")
+        unsigned_link = job.get_results_metadata_url(full=True)
+
+        cube = con120.load_stac_from_job(job)
+        fg = cube.flat_graph()
+
+        assert fg == {
+            "loadstac1": {
+                "process_id": "load_stac",
+                "arguments": {"url": unsigned_link},
```

Comment:
I think your expected value should not depend on get_results_metadata_url here, but just explicitly state `"https://oeo.test/jobs/j0bi6/results"`